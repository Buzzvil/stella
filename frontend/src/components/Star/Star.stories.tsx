import React from "react";
import Styled from "styled-components";

import { storiesOf } from "@storybook/react";

import Star from "./Star";

const SizedContainer = Styled.div`
  font-size: 30px;
`;

storiesOf("Components|Star", module)
  .add("Empty", () => <Star value={0} />)
  .add("Half", () => <Star value={0.5} />)
  .add("Filled", () => <Star value={1} />)
  .add("Resized", () => (
    <SizedContainer>
      <Star value={1} />
    </SizedContainer>
  ));
