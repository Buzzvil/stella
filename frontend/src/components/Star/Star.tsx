import React from "react";
import Styled from "styled-components";

import StarEmpty from "../../img/Star-empty.svg";
import StarHalf from "../../img/Star-half.svg";
import StarFilled from "../../img/Star-filled.svg";

interface StarProps {
  value: number;
}

const StarImg = Styled.img`
  height: 1em;
`;

const Star: React.SFC<StarProps> = ({ value }) => {
  const img = value >= 1 ? StarFilled : value >= 0.5 ? StarHalf : StarEmpty;
  return <StarImg src={img} />;
};

export default Star;
