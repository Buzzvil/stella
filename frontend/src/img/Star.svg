<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16.7827 21.34C16.5227 21.34 16.2527 21.29 16.0027 21.18L12.7927 19.82C12.2927 19.61 11.7327 19.61 11.2327 19.82L8.02267 21.18C7.37267 21.46 6.64266 21.37 6.06266 20.96C5.49266 20.54 5.18266 19.87 5.24266 19.17L5.54266 15.69C5.59266 15.15 5.41266 14.61 5.06266 14.21L2.77267 11.57C2.30267 11.04 2.16267 10.32 2.37267 9.65002C2.58267 8.98002 3.13267 8.48002 3.82267 8.32002L7.22266 7.53002C7.74266 7.41002 8.20267 7.07002 8.48267 6.61002L10.2827 3.62002C10.6527 3.01002 11.2927 2.65002 12.0027 2.65002C12.7127 2.65002 13.3527 3.01002 13.7127 3.62002L15.5127 6.61002C15.7927 7.07002 16.2527 7.40002 16.7727 7.53002L20.1727 8.32002C20.8627 8.48002 21.4027 8.98002 21.6227 9.65002C21.8427 10.32 21.6927 11.04 21.2327 11.58L18.9427 14.21C18.5927 14.62 18.4127 15.16 18.4627 15.69L18.7627 19.17C18.8227 19.88 18.5227 20.55 17.9427 20.96C17.6027 21.21 17.1927 21.34 16.7827 21.34ZM12.0027 17.67C12.5327 17.67 13.0627 17.78 13.5627 17.99L16.7727 19.35L16.4727 15.87C16.3827 14.8 16.7327 13.71 17.4327 12.9L19.7227 10.27L16.3227 9.48002C15.2827 9.23002 14.3627 8.57002 13.8027 7.64002L12.0027 4.64002L10.2027 7.64002C9.64266 8.56002 8.72266 9.23002 7.67266 9.47002L4.28267 10.26L6.57267 12.89C7.28267 13.7 7.63267 14.78 7.53267 15.86L7.23267 19.34L10.4427 17.98C10.9427 17.77 11.4727 17.67 12.0027 17.67Z" fill="#E0E0E0"/>
<mask id="mask0" mask-type="alpha" maskUnits="userSpaceOnUse" x="2" y="2" width="20" height="20">
<path d="M10.2873 3.62156C11.0642 2.32992 12.9358 2.32992 13.7127 3.62156L15.5111 6.61155C15.7902 7.07557 16.2454 7.40651 16.7728 7.5287L20.1706 8.31608C21.6384 8.65621 22.2168 10.4372 21.2291 11.5756L18.9427 14.2109C18.5879 14.6198 18.414 15.1553 18.4608 15.6948L18.7624 19.1715C18.8926 20.6733 17.3785 21.774 15.9912 21.1859L12.7798 19.8246C12.2814 19.6134 11.7186 19.6134 11.2202 19.8246L8.00879 21.1859C6.62148 21.774 5.10735 20.6733 5.23762 19.1715L5.53918 15.6948C5.58598 15.1553 5.41209 14.6198 5.05727 14.2109L2.77092 11.5756C1.78325 10.4372 2.36159 8.65621 3.82942 8.31608L7.22724 7.5287C7.75456 7.40651 8.20981 7.07557 8.48891 6.61155L10.2873 3.62156Z" fill="#FFE76A"/>
</mask>
<g mask="url(#mask0)">
<rect width="24" height="24" fill="#FFE76A"/>
</g>
</svg>
