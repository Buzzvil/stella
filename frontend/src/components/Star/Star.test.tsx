import React from "react";
import renderer from "react-test-renderer";

import Star from "./Star";

it("renders single svg element", () => {
  const render = renderer.create(<Star value={0} />);
  const elements = render.root.findAllByType("svg");
  expect(elements.length).toEqual(1);
});

it("displays with empty image when value is 0", () => {
  const render = renderer.create(<Star value={0} />);
  const svg = render.root.findByType("svg");
  expect(svg.props.className).toMatchInlineSnapshot(`"sc-bwzfXH iHgQBh"`);
});

it("displays with half filled image when value is .5", () => {
  const render = renderer.create(<Star value={0.5} />);
  const svg = render.root.findByType("svg");
  expect(svg.props.className).toMatchInlineSnapshot(`"sc-bwzfXH kuAcAi"`);
});

it("displays with filled image when value is 1", () => {
  const render = renderer.create(<Star value={1} />);
  const svg = render.root.findByType("svg");
  expect(svg.props.className).toMatchInlineSnapshot(`"sc-bwzfXH eXPEPj"`);
});
