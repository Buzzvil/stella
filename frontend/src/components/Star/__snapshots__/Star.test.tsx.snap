// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays with empty image when value is .5 1`] = `
<img
  className="sc-bdVaJa kCvZfg"
  src="Star-half.svg"
/>
`;

exports[`displays with empty image when value is 0 1`] = `
<img
  className="sc-bdVaJa kCvZfg"
  src="Star-empty.svg"
/>
`;

exports[`displays with empty image when value is 1 1`] = `
<img
  className="sc-bdVaJa kCvZfg"
  src="Star-filled.svg"
/>
`;
